Assets/Scripts/UI/InGameActions.cs

```diff
@@ -40,11 +49,17 @@ private void OnEnable()
         restart.Raise();
     }
     private void OnClickPause() {
-        isPausedDisplayed.Value = !isPausedDisplayed.Value;
-        togglePause.Raise();
+       if (levelState.Machine.Value == LevelStates.InProgress){
+            levelState.Machine.Dispatch(LevelTransition.Pause);
+            GetComponent<PauseActions>().OnPause();

```

Comment:
~I'm asking myself whether we need a state machine in that place after all~

Reply:
~I think we should clearly separate UI logic from pause itself. IMO we should raise an event, and something closer to the game itself should handle the Pause, without any link between the two~  
Nevermind you did that already :+1: